### PR TITLE
Display the packages that having deprecated dependencies

### LINKED ISSUE #88: Display the deprecated dependency for each package in the dashboard

### PR DESCRIPTION
**Pull request by liyanhui1228**
Fixes #88

### CODE REVIEW

File: dashboard/grid-template.html [hunk 1]
```diff
@@ -117,7 +117,9 @@
     {% set row = loop.index0 %}
     <tr>
         <td><font face="verdana">{{ row_package.friendly_name }}</font>{% if results.has_issues(row_package) %}
-                <img class="warning-img" src="https://www.gstatic.com/images/icons/material/system/1x/warning_amber_18dp.png">
+                {% set issues = results.deprecated_deps[row_package.friendly_name][0] %}
+                <img class="warning-img" class="dialog" onclick="showIssues('{{ row_package.friendly_name|safe }}', '{{ issues|replace('\'', '\\\'')|safe }}')" src="https://www.gstatic.com/images/icons/material/system/1x/warning_amber_18dp.png">
```
Comment (brianquinlan) [inline dashboard/grid-template.html]: Can't jinja2 do this escaping for you? I'm not sure.

File: dashboard/grid_builder.py [hunk 2]
```diff
@@ -88,8 +96,32 @@ def has_issues(self, p: package.Package) -> bool:
...
+    def get_deprecated_deps(self):
+        """
+        Returns True if there are deprecated dependencies for a
+        given package.
+        """
+        finder = deprecated_dep_finder.DeprecatedDepFinder()
+        deprecated_deps = list(finder.get_deprecated_deps())
+
+        results = {}
+        for item in deprecated_deps:
```
Comment (brianquinlan) [inline dashboard/grid_builder.py]: Could you use tuple unpacking so that item[0][X] has a name?

File: dashboard/grid_builder.py [hunk 3]
```diff
@@ -88,8 +96,32 @@ def has_issues(self, p: package.Package) -> bool:
...
+
+            # Whether the package has deprecated dependencies or not
+            if self.deprecated_deps[p.install_name][1]:
+                return True
+
         return False
 
+    def get_deprecated_deps(self):
+        """
+        Returns True if there are deprecated dependencies for a
```
Comment (brianquinlan) [inline dashboard/grid_builder.py]: Maybe add the type annotation? This doesn't return a bool.

File: dashboard/grid_builder.py (hunk 2 above)
Comment (liyanhui1228) [reply dashboard/grid_builder.py]: Done.

File: dashboard/grid-template.html (hunk 1 above)
Comment (liyanhui1228) [reply dashboard/grid-template.html]: Removed, jinja2 couldn't do the escaping for `'`, so I have to manually escape it here.

File: dashboard/grid_builder.py (hunk 3 above)
Comment (liyanhui1228) [reply dashboard/grid_builder.py]: Done.

**Pull request merged**